### PR TITLE
Update dump_kubernetes.sh

### PR DESCRIPTION
**Pull request by Tahler**
- Add some resiliency
  - Add `mkdir -p`
  - Only request specific resources if they exist
- Dumps miscellaneous info to OUT_DIR not LOG_DIR
- Copies core dumps of all istio-proxy containers if present
- Adds flag --error-if-nasty-logs which errors if logs match a grep

### CODE REVIEW

**Comment by mandarjog** [comment]
@kyessenov can you please review this? Thanks.

File: bin/dump_kubernetes.sh [hunk 1]
```diff
@@ -73,7 +84,64 @@ dump_time() {
...
+    core_dumps=$(kubectl exec -n "${namespace}" "${pod}" -c "${container}" -- \
+        find /etc/istio/proxy -name 'core.*')
+    for f in ${core_dumps}; do
+      local out_file
+      out_file="${out_dir}/$(basename "${f}")"
+
+      kubectl exec -n "${namespace}" "${pod}" -c "${container}" -- \
+          cat "${f}" > "${out_file}"
+
+      log "Copied ${namespace}/${pod}/${container}:${f} to ${out_file}"
```
Comment (kyessenov) [inline bin/dump_kubernetes.sh]: cc @ostromart to confirm the paths to core dumps.

File: bin/dump_kubernetes.sh [hunk 2]
```diff
@@ -86,54 +154,51 @@ dump_logs() {
...
 
-dump_resources() {
-  log "Retrieving resource configurations"
+dump_kubernetes_resources() {
+  log "Retrieving kubernetes resource configurations"
 
   mkdir -p "${OUT_DIR}"
   # Only works in Kubernetes 1.8.0 and above.
   kubectl get --all-namespaces --export \
       all,ingresses,endpoints,customresourcedefinitions,configmaps,secrets,events \
```
Comment (kyessenov) [inline bin/dump_kubernetes.sh]: add `jobs`

File: bin/dump_kubernetes.sh [hunk 3]
```diff
@@ -73,7 +84,64 @@ dump_time() {
...
+  local namespace="${1}"
+  local pod="${2}"
+  local container="${3}"
+
+  if [ "istio-proxy" = "${container}" ]; then
+    local out_dir="${LOG_DIR}/${namespace}/${pod}"
+    mkdir -p "${out_dir}"
+    local core_dumps
+    core_dumps=$(kubectl exec -n "${namespace}" "${pod}" -c "${container}" -- \
+        find /etc/istio/proxy -name 'core.*')
```
Comment (ostromart) [inline bin/dump_kubernetes.sh]: please also add the new path /var/istio/proxy

Comment (Tahler) [reply bin/dump_kubernetes.sh]: Done.

File: bin/dump_kubernetes.sh (hunk 2 above)
Comment (Tahler) [reply bin/dump_kubernetes.sh]: Done.

**Pull request merged**